datanode/service/market.go

```diff
@@ -46,6 +50,28 @@ func NewMarkets(store MarketStore, log *logging.Logger) *Markets {
 }
 
 func (m *Markets) Initialise(ctx context.Context) error {
+	// This hacky timer is needed to ensure that the cache is re-populated when restoring
+	// from network history segments. It would be better to explicitly refresh the cache when
+	// loading in a segment, but this quick fix will work for now.
+	go func() {
+		ticker := time.NewTicker(refreshCacheEvery)
+		for {
+			select {
+			case <-ticker.C:
+				{
+					err := m.refreshCache(ctx)
+					if err != nil {
+						m.log.Warn("error refreshing market cache", logging.Error(err))
+					}
+				}
+			}
+		}
```

Comment:
Could be simplified a little.
```suggestion
		for range time.NewTicker(refreshCacheEvery).C {
			if err := m.refreshCache(ctx); err != nil {
				m.log.Warn("error refreshing market cache", logging.Error(err))
			}
		}
```

Reply:
Roger wilco

Reply:
(this is done)